tests/components/openerz/test_sensor.py

```diff
@@ -0,0 +1,75 @@
+"""Tests for OpenERZ component."""
+from copy import deepcopy
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.setup import async_setup_component
+
+MOCK_CONFIG = {
+    "sensor": {
+        "platform": "openerz",
+        "name": "test_name",
+        "zip": 1234,
+        "waste_type": "glass",
+    }
+}
+
+
+async def test_sensor_init(hass):
+    """Test whether all values initialized properly."""
+    with patch(
+        "homeassistant.components.openerz.sensor.OpenERZSensor.update",
+        return_value=True,
+    ) as patched_update:
+        await async_setup_component(hass, SENSOR_DOMAIN, MOCK_CONFIG)
+
+        entity_id = "sensor.test_name"
+        test_openerz = hass.data[SENSOR_DOMAIN].get_entity(entity_id)
```

Comment:
We shouldn't interact with the entity directly at all, only the state. The state is the truth of the entity.